src/collections/Form/Form.js

```diff
@@ -1,143 +1,261 @@
-import $ from 'jquery'
 import _ from 'lodash'
-import React, { Component, PropTypes } from 'react'
+import React, { PropTypes } from 'react'
 
 import {
-  deprecateProps,
   getElementType,
   getUnhandledProps,
+  makeDebugger,
   META,
+  SUI,
+  useKeyOnly,
+  useWidthProp,
 } from '../../lib'
+import FormButton from './FormButton'
+import FormCheckbox from './FormCheckbox'
+import FormDropdown from './FormDropdown'
 import FormField from './FormField'
-import FormFields from './FormFields'
-
-export default class Form extends Component {
-  static propTypes = {
-    /** An element type to render as (string or function). */
-    as: PropTypes.oneOfType([
-      PropTypes.string,
-      PropTypes.func,
-    ]),
-
-    children: PropTypes.node,
-    className: PropTypes.string,
-    settings: PropTypes.shape({
-      on: PropTypes.string,
-      inline: PropTypes.bool,
-      fields: PropTypes.object,
-    }),
-
-    // form settings
-    keyboardShortcuts: PropTypes.bool,
-    on: PropTypes.oneOf([
-      'blur',
-      'change',
-      'submit',
-    ]),
-    revalidate: PropTypes.bool,
-    delay: PropTypes.bool,
-    inline: PropTypes.bool,
-    transition: PropTypes.string,
-    duration: PropTypes.number,
-    // callbacks
-    onValid: PropTypes.func,
-    onInvalid: PropTypes.func,
-    onSuccess: PropTypes.func,
-    onFailure: PropTypes.func,
-    fields: PropTypes.object,
-  }
+import FormGroup from './FormGroup'
+import FormInput from './FormInput'
+import FormRadio from './FormRadio'
+import FormSelect from './FormSelect'
+import FormTextArea from './FormTextArea'
 
-  static defaultProps = {
-    as: 'form',
-    // prevent submit by default
-    // https://github.com/Semantic-Org/Semantic-UI/issues/546
-    onSuccess: () => false,
-  }
+const debug = makeDebugger('form')
 
-  constructor(props, context) {
-    super(props, context)
-    deprecateProps(this, {
-      settings: 'Use a separate prop for each setting.',
-    })
-  }
+let getNodeName = ({ name }) => name
+const debugSerializedResult = (json, name, node) => {
+  debug('serialized ' + JSON.stringify({ [name]: json[name] }) + ' from:', node)
+}
 
-  componentDidMount() {
-    this.refresh()
-  }
+// in dev, warn about form nodes missing a "name"
+if (process.NODE_ENV !== 'production') {
+  getNodeName = (node) => {
+    const name = node.name
 
-  componentDidUpdate(prevProps, prevState) {
-    this.refresh()
-  }
+    if (!name) {
+      const errorMessage = [
+        'Encountered a form control node without a name attribute.',
+        'Each node in a group should have a name.',
+        'Otherwise, the node will serialize as { "undefined": <value> }.',
+      ].join(' ')
+      console.error(errorMessage, node) // eslint-disable-line no-console
+    }
 
-  componentWillUnmount() {
-    _.invoke(this, 'element.off')
+    return name
   }
+}
 
-  static _meta = {
-    name: 'Form',
-    type: META.TYPES.COLLECTION,
-  }
+function formSerializer(formNode) {
```

Comment:
Heads up, this serializer reads from the DOM.  This is a React antipattern, though short of the validation PR which will handle input binding, this is how we're achieving serialization.